File: apps/android/app/src/main/kotlin/sh/nikhil/swekitty/ui/MemoryButton.kt
package sh.nikhil.swekitty.ui

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Article
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.runtime.Composable

/**
 * Header affordance that flips the Browser tab between the live preview
 * and the session memory HTML at <endpoint>/memory/sessions/<uuid>.html.
 * Tapping switches to the Browser tab (via [onJumpToBrowser]) and toggles
 * the mode; tapping again returns to preview.
 */
@Composable
fun MemoryButton(
    currentMode: BrowserMode,
    onToggle: (BrowserMode) -> Unit,
    onJumpToBrowser: () -> Unit,
) {
    IconButton(onClick = {
        onToggle(if (currentMode == BrowserMode.Memory) BrowserMode.Preview else BrowserMode.Memory)
        onJumpToBrowser()
    }) {
        Icon(Icons.Outlined.Article, contentDescription = "Memory")
    }
}

File: apps/ios/Sources/Views/MemoryButton.swift
import SwiftUI

/// Header affordance that flips the Browser tab between the live `preview`
/// and the session memory HTML at `<endpoint>/memory/sessions/<uuid>.html`.
/// Tapping switches into the Browser tab and toggles the mode; tapping
/// again with the tab already on memory flips back to preview.
struct MemoryButton: View {
    @Binding var tab: ProjectTab
    @Binding var mode: BrowserMode

    var body: some View {
        Button {
            tab = .browser
            mode = (mode == .memory) ? .preview : .memory
        } label: {
            Image(systemName: mode == .memory ? "doc.text.fill" : "doc.text")
        }
        .accessibilityLabel("Open session memory")
    }
}
